src/Xamarin.Android.Build.Tasks/Tasks/LinkAssembliesNoShrink.cs

```diff
@@ -117,5 +117,28 @@ public override void LogMessage (string message)
 				logger.LogDebugMessage ("{0}", message);
 			}
 		}
+
+		class AddKeepAlivesStep : MonoDroid.Tuner.AddKeepAlivesStep
+		{
+			readonly DirectoryAssemblyResolver resolver;
+			readonly TaskLoggingHelper logger;
+
+			public AddKeepAlivesStep (DirectoryAssemblyResolver resolver, TypeDefinitionCache cache, TaskLoggingHelper logger)
+				: base (cache)
+			{
+				this.resolver = resolver;
+				this.logger = logger;
+			}
+
+			protected override AssemblyDefinition GetCorlibAssembly ()
+			{
+				return resolver.GetAssembly ("mscorlib.dll");
```

Comment:
When this task runs in .NET 6 does it need to look for `System.Private.CoreLib`?

We can't use `#if` here, this assembly is always `netstandard2.0`. You might need to pass `$(UsingAndroidNETSdk)` into this task, which is `true` in .NET 6 and blank otherwise.

Reply:
Good tip about the `netstandard2.0`. It took me some time to notice that :-) I have added detection of `System.Private.CoreLib` during the resolver preparation, that should work too.